src/test/java/StringManipulationTest.java

```diff
@@ -19,63 +18,113 @@ public void tearDown() {
         manipulatedstring = null;
     }
 
+    @Test
+    public void testgetString(){
+        manipulatedstring.setString("Hello!");
+        String expected = "Hello!";
+        String actual = manipulatedstring.getString();
+        assertEquals(expected, actual);
+    }
     @Test
     public void testCount1() {
         manipulatedstring.setString("This is my string");
         int length = manipulatedstring.count();
         assertEquals(4, length);
     }
 
+    // Test when string has multiple punctuations.
     @Test
     public void testCount2() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("This isn't my string!");
+        int length = manipulatedstring.count();
+        assertEquals(4, length);
     }
 
+    // Test when string is empty.
+    // Expected return is 0
     @Test
     public void testCount3() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("");
+        int length = manipulatedstring.count();
+        assertEquals(0, length);
     }
 
+    // Test when string is null
```

Comment:
I like that you added comments describing a bit about the test. Try looking into the @DisplayName annotation for an addition nice way to format test details

Reply:
Oh, that's really cool. I'll use that